src/cascadia/LocalTests_TerminalApp/TabTests.cpp

```diff
@@ -164,4 +176,325 @@ namespace TerminalAppLocalTests
         VERIFY_SUCCEEDED(result);
     }
 
+    void TabTests::CreateTerminalPage()
+    {
+        winrt::com_ptr<winrt::TerminalApp::implementation::TerminalPage> page{ nullptr };
+
+        auto result = RunOnUIThread([&page]() {
+            page = winrt::make_self<winrt::TerminalApp::implementation::TerminalPage>();
+            VERIFY_IS_NOT_NULL(page);
+        });
+        VERIFY_SUCCEEDED(result);
+    }
+
+    // Method Description:
+    // - This is a helper to set up a TerminalPage for a unittest. This method
+    //   does a couple things:
+    //   * Create()'s a TerminalPage with the given settings. Constructing a
+    //     TerminalPage so that we can get at its implementation is wacky, so
+    //     this helper will do it correctly for you, even if this doesn't make a
+    //     ton of sense on the surface. This is also why you need to pass both a
+    //     projection and a com_ptr to this method.
+    //   * It will use the provided settings object to initialize the TerminalPage
+    //   * It will add the TerminalPage to the test Application, so that we can
+    //     get actual layout events. Much of the Terminal assumes there's a
+    //     non-zero ActualSize to the Terminal window, and adding the Page to
+    //     the Application will make it behave as expected.
+    //   * It will wait for the TerminalPage to finish initialization before
+    //     returning control to the caler. It does this by creating an event and
+    //     only setting the event when the TerminalPage raises its Initialized
+    //     event, to signal that startup is complete. At this point, there will
+    //     be one tab with the default profile in the page.
+    //   * It will also ensure that the first tab is focused, since that happens
+    //     asynchronously in the application typically.
+    // Arguments:
+    // - projectedPage: a TerminalPage projected type that will recieve the new TerminalPage instance

```

Comment:
projected type can be trivially generated from the com_ptr one; do we need both?

Reply:
Okay yea I thought I might still need that for wackiness's sake. Removed